src/TreeHouse/Queue/Message/Provider/AmqpMessageProvider.php

```diff
@@ -51,4 +51,20 @@ public function nack(Message $message, $requeue = false)
     {
         $this->queue->nack($message->getId(), $requeue ? AMQP_REQUEUE : null);
     }
+
+    /**
+     * @inheritdoc
+     */
+    public function consume(callable $callback)
+    {
+        $this->queue->consume(function(\AMQPEnvelope $envelope) use ($callback) {
+            $id    = $envelope->getDeliveryTag();
+            $body  = $envelope->getBody();
+            $props = new MessageProperties($envelope->getHeaders());
+
+            $message = new Message($body, $props, $id);
+
+            $callback($message);
```

Comment:
I would add support here to bail out of the consume process, when the callback returns false: http://php.net/manual/pl/amqpqueue.consume.php